eoscompanion/routes.py

```diff
@@ -190,7 +194,7 @@ def _callback(applications):
                     'icon': format_app_icon_uri(a.icon, query['deviceUUID']),
                     'language': a.language
                 }
-                for a in applications
+                for a in filtered_applications
```

Comment:
You can also move the check down here ;-)

I can't see from the face of it why it wouln't be working